app/src/test/java/com/kickstarter/viewmodels/RewardsFragmentViewModelTest.kt

```diff
@@ -331,19 +343,121 @@ class RewardsFragmentViewModelTest : KSRobolectricTestCase() {
 
         // - We check that the viewModel has filtered out the rewards not started yet
         this.projectData.assertValue(modifiedPData)
-        this.rewardsCount.assertValue(5)
     }
 
     @Test
-    fun testRewardsCount() {
-        val project = ProjectFactory.project()
+    fun `test countrySelectorRules state contains appropriate ShippingRules when reward shipping worldwide and default location Canada`() = runTest {
+
+        val unlimitedReward = RewardFactory.rewardWithShipping()
+
+        val rewards = listOf<Reward>(
+            unlimitedReward
+        )
+        val project = ProjectFactory.project().toBuilder().rewards(rewards).build()
+
+        val config = ConfigFactory.configForCA()
+        val currentConfig = MockCurrentConfigV2()
+        currentConfig.config(config)
+
+        val testShippingRulesList = ShippingRulesEnvelopeFactory.shippingRules()
+        val apolloClient = object : MockApolloClientV2() {
+            override fun getShippingRules(reward: Reward): Observable<ShippingRulesEnvelope> {
+                return Observable.just(testShippingRulesList)
+            }
+        }
+
+        val user = UserFactory.user()
+        val env = environment()
             .toBuilder()
-            .rewards(listOf(RewardFactory.noReward(), RewardFactory.reward()))
+            .currentUserV2(MockCurrentUserV2(user))
+            .apolloClientV2(apolloClient)
+            .currentConfig2(currentConfig)
             .build()
+
+        val state = mutableListOf<ShippingRulesState>()
+        val dispatcher = UnconfinedTestDispatcher(testScheduler)
+        backgroundScope.launch(dispatcher) {
```

Comment:
Testing `GetShippingRulesUseCase` requires co-routines, on future PR's (probably [here](https://kickstarter.atlassian.net/browse/MBL-1576)), once the rewards are filtered by location will likely have a testfile specific to the useCase